cli/tests/upload.rs

```diff
@@ -72,17 +72,23 @@ async fn upload_bundle() {
         })
     );
 
-    assert_eq!(
-        requests_iter.next().unwrap(),
-        RequestPayload::CreateBundleUpload(CreateBundleUploadRequest {
-            repo: Repo {
+    let request = requests_iter.next().unwrap();
+    if let RequestPayload::CreateBundleUpload(upload_request) = request {
+        assert_eq!(
+            upload_request.repo,
+            Repo {
                 host: String::from("github.com"),
                 owner: String::from("trunk-io"),
                 name: String::from("analytics-cli"),
-            },
-            org_url_slug: String::from("test-org"),
-        })
-    );
+            }
+        );
+        assert_eq!(upload_request.org_url_slug, String::from("test-org"));
+        assert!(upload_request.client_version.starts_with("trunk-analytics-cli cargo="));
+        assert!(upload_request.client_version.contains(" git="));
+        assert!(upload_request.client_version.contains(" rustc="));
+    } else {
+        panic!("Expected CreateBundleUpload request, got {:?}", request);
+    }
```

Comment:
nit: You can use the fancy `assert_matches` macro to do the failure for when the pattern isn't matched here

```suggestion
    let upload_request =
        assert_matches!(requests_iter.next().unwrap(), RequestPayload::CreateBundleUpload(ur) => ur);
    assert_eq!(
        upload_request.repo,
        Repo {
            host: String::from("github.com"),
            owner: String::from("trunk-io"),
            name: String::from("analytics-cli"),
        }
    );
    assert_eq!(upload_request.org_url_slug, String::from("test-org"));
    assert!(upload_request.client_version.starts_with("trunk-analytics-cli cargo="));
    assert!(upload_request.client_version.contains(" git="));
    assert!(upload_request.client_version.contains(" rustc="));
```